test/unit/test_keyring_multi.py

```diff
@@ -117,23 +137,39 @@ def test_keyring_with_no_generator_no_children():
     assert exc_info.match("At least one of generator or children must be provided")
 
 
-def test_children_not_keyrings():
+@pytest.mark.parametrize(
+    "generator, children",
+    (
+        (WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING, None),
+        (WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING, get_multi_keyring_with_no_generator().children),
+        (None, [WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING]),
+        (get_multi_keyring_with_no_children().generator, [WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING]),
+    ),
+)
+def test_keyring_with_invalid_parameters(generator, children):
     with pytest.raises(TypeError) as exc_info:
-        MultiKeyring(children=[WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING])
-    assert exc_info.errisinstance(TypeError)
+        MultiKeyring(generator=generator, children=children)
+    assert exc_info.match("('children'|'generator') must be <class 'aws_encryption_sdk.keyring.base.Keyring'>.*")
+
+
+def test_decryption_keyrings():
+    test_multi_keyring = get_multi_keyring_with_generator_and_children()
+    assert test_multi_keyring.generator in test_multi_keyring._decryption_keyrings
+    for child_keyring in test_multi_keyring.children:
+        assert child_keyring in test_multi_keyring._decryption_keyrings
```

Comment:
Let's also add an assertion that there are no additional keyrings in `_decryption_keyrings`

```python
assert len(test_multi_keyring._decryption_keyrings) == len(test_multi_keyring.children) + 1
```